app/pods/agendas/controller.js

```diff
@@ -2,15 +2,45 @@ import Controller from '@ember/controller';
 import { tracked } from '@glimmer/tracking';
 import { inject as service } from '@ember/service';
 import { action } from '@ember/object';
-// TODO: determine if these default qp's are still needed here, otherwise refactor to mixin-less solution
-// eslint-disable-next-line ember/no-mixins
-import DefaultQueryParamsMixin from 'ember-data-table/mixins/default-query-params';
+import { restartableTask, timeout } from 'ember-concurrency';
 
-export default class AgendasController extends Controller.extend(DefaultQueryParamsMixin) {
+export default class AgendasController extends Controller {
+  queryParams = ['page', 'size', 'sort', 'filter'];
+
+  @service router;
   @service currentSession;
   @service router;
 
+  @tracked isLoadingModel = false;
   @tracked isCreatingNewSession = false;
+  @tracked filter = null;
+  @tracked page = 0;
+  @tracked size = 10;
+  @tracked sort = 'created-for.is-final,-created-for.planned-start,created-for.kind.label';
+
+  dateRegex = /^(?:\d{1,2}\/)??(?:\d{1,2}\/)?\d{4}$/;
+
+  @restartableTask
+  *debouncedSetFilter(event) {
+    yield timeout(500);
+    this.setFilter(event.target.value);
+  }
+
+  @action
+  setFilter(date) {
+    if (this.dateRegex.test(date)) {
+      this.filter = date;
+    } else if (date === '') {
+      this.filter = null;
+    }
+    this.page = 0;
```

Comment:
You only want to reset the page to 0 if `this.filter` actually changed (so in both if-branches). In the current implementation, if you're not on the first page and you start typing slowly in the search field, page is reset to 0 (and filter cleared) before you finish entering the date.